src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1317,13 +1317,22 @@ public static SqlConnectionStringBuilder CreateConnectionStringBuilder(Connectio
             {
                 string connectionString = string.Empty;
                 ConnectionInfo info;
-                if (TryFindConnection(connStringParams.OwnerUri, out info))
+                SqlConnectionStringBuilder connStringBuilder;
+                if (connStringParams.OwnerUri == null)
+                {
+                    // Use connection info to create connection string
+                    if (connStringParams.ConnectionDetails != null)
+                    {
+                        connStringBuilder = CreateConnectionStringBuilder(connStringParams.ConnectionDetails);

```

Comment:
You still want this to use the same logic as below for checking includePassword and includeApplicationName. That's all common to either way so just first get the connStringBuilder based on what type of connection info we're given and then run the common logic.